packages/appium/lib/cli/extension-command.js

```diff
@@ -513,7 +526,12 @@ class ExtensionCommand {
         `Can't uninstall ${this.type} '${installSpec}'; it is not installed`
       );
     }
-    const pkgName = this.config.installedExtensions[installSpec].pkgName;
+    const extRecord = this.config.installedExtensions[installSpec];
+    if (extRecord.installType === INSTALL_TYPE_DEV) {
+      this.log.warn(`Cannot uninstall ${this.type} "${installSpec}" because it is in development!`);
```

Comment:
I don't quite understand the scenario here. Let say I am developing a driver. I checked out the sources and then installed them to appium server. Tested it, everything is fine, then published. Now I want to switch the driver to the NPM branch to verify all works as expected. How do I do it if the dev version cannot be uninstalled (I'm not talking about removing sources physically, but rather removing it from the manifest)

Reply:
i'm pretty sure that 'dev' extensions are not actually 'installed'. they do not live in an appium manifest anywhere. they're dynamically detected based on the current directory. the way to 'uninstall' them is to simply move out of the development directory.

Reply:
You use a different directory, I guess?  "dev mode" only applies if `CWD` is the directory of an extension (or a subdirectory thereof), _and_ that extension depends on `appium` (as it should).  It doesn't make sense to "uninstall" something that way.  If you want to test your _published_ package, you will need to specify a different `APPIUM_HOME`.  Just like you wouldn't `npm install appium-xcuitest-driver` _from the `appium-xcuitest-driver` working copy_.